tests/test-server/src/main.rs

```diff
@@ -35,6 +35,7 @@ async fn main() -> std::io::Result<()> {
     let port = args.port.unwrap_or(53);
     let transport = args.transport.unwrap_or("both".into());
     let handler = match args.handler {
+        Some(handler) if handler == "bad_txid" => handlers::bad_txid_handler,
```

Comment:
I think you can do `Some("bad_txid")` directly if you coerce `args.handler` to `&str`, maybe with `&` or `&*` or `as_str()`?